lib/binding_web/binding.js

```diff
@@ -842,78 +842,90 @@ class Language {
           const operator = steps[0].value;
           let isPositive = true;
           let matchAll = true;
+          let captureName;
           switch (operator) {
             case 'any-not-eq?':
-              isPositive = false;
-              matchAll = false;
-            case 'any-eq?':
-              matchAll = false;
             case 'not-eq?':
               isPositive = false;
+            case 'any-eq?':
             case 'eq?':
               if (steps.length !== 3) throw new Error(
-                `Wrong number of arguments to \`#eq?\` predicate. Expected 2, got ${steps.length - 1}`
+                `Wrong number of arguments to \`#${operator}\` predicate. Expected 2, got ${steps.length - 1}`
               );
               if (steps[1].type !== 'capture') throw new Error(
-                `First argument of \`#eq?\` predicate must be a capture. Got "${steps[1].value}"`
+                `First argument of \`#${operator}\` predicate must be a capture. Got "${steps[1].value}"`
               );
+              let eqMatchAll = !operator.startsWith('any-');
+              let eqIsPositive = isPositive;
               if (steps[2].type === 'capture') {
                 const captureName1 = steps[1].name;
                 const captureName2 = steps[2].name;
-                textPredicates[i].push(function(captures) {
+                textPredicates[i].push(function (captures) {
                   let nodes_1 = [];
                   let nodes_2 = [];
                   for (const c of captures) {
                     if (c.name === captureName1) nodes_1.push(c.node);
                     if (c.name === captureName2) nodes_2.push(c.node);
                   }
-                  return matchAll
-                    ? nodes_1.every(n1 => nodes_2.some(n2 => n1.text === n2.text)) === isPositive
-                    : nodes_1.some(n1 => nodes_2.some(n2 => n1.text === n2.text)) === isPositive;
```

Comment:
We can't do it this way because the logic inverts twice: first between `eq` and `not-eq`, then between both those predicates and their `any-` variants.

Hard to explain in words, but the `any-not-eq?` case falls down because it tries to assert that “some of these nodes match X” is `false`, when it needs to assert that “some of these nodes _don't_ match X” is `true`.